### PR TITLE
[4.0] Add prepared statements for mod_articles_archive

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the module in all ways you can think of.

* create a frontend module
* create articles with the condition archvied
* The module should show you the articles as without the PR

### Expected result
Nothing changed.

### CODE REVIEW

File: modules/mod_articles_archive/Helper/ArticlesArchiveHelper.php [hunk 1]
```diff
@@ -35,30 +36,29 @@ class ArticlesArchiveHelper
...
-			->innerJoin($db->quoteName('#__workflow_associations', 'wa') . ' ON wa.item_id = c.id')
-			->innerJoin($db->quoteName('#__workflow_stages', 'ws') . ' ON wa.stage_id = ws.id')
-			->where($db->quoteName('ws.condition') . ' = ' . (int) ContentComponent::CONDITION_ARCHIVED)
+			->innerJoin($db->quoteName('#__workflow_associations', 'wa'), $db->quoteName('wa.item_id') . ' = ' . $db->quoteName('c.id'))
+			->innerJoin($db->quoteName('#__workflow_stages', 'ws'), $db->quoteName('wa.stage_id') . ' = ' . $db->quoteName('ws.id'))
+			->where($db->quoteName('ws.condition') . ' = :condition')
 			->group($query->year($db->quoteName('c.created')) . ', ' . $query->month($db->quoteName('c.created')))
-			->order($query->year($db->quoteName('c.created')) . ' DESC, ' . $query->month($db->quoteName('c.created')) . ' DESC');
+			->order($query->year($db->quoteName('c.created')) . ' DESC, ' . $query->month($db->quoteName('c.created')) . ' DESC')
+			->bind(':condtion', $condition, ParameterType::INTEGER);
```
Comment (Quy) [inline modules/mod_articles_archive/Helper/ArticlesArchiveHelper.php]:
```suggestion
			->bind(':condition', $condition, ParameterType::INTEGER);
```

File: modules/mod_articles_archive/Helper/ArticlesArchiveHelper.php [hunk 2]
```diff
@@ -35,30 +36,29 @@ class ArticlesArchiveHelper
 	 */
 	public static function getList(&$params)
 	{
-		// Get application
-		$app = Factory::getApplication();
+		$app       = Factory::getApplication();
+		$db        = Factory::getDbo();
+		$condition = ContentComponent::CONDITION_ARCHIVED;
```
Comment (Quy) [inline modules/mod_articles_archive/Helper/ArticlesArchiveHelper.php:41]: Remove variable.

Comment (HLeithner) [reply modules/mod_articles_archive/Helper/ArticlesArchiveHelper.php:41]: This will be removed anyway when workflow changes more into a state machine

**Review by alikon — APPROVED** [review]

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 1b6b1ef2668bc7b3e74ba1672551165248bfc691<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25041">issues.joomla.org/tracker/joomla-cms/25041</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1b6b1ef2668bc7b3e74ba1672551165248bfc691<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25041">issues.joomla.org/tracker/joomla-cms/25041</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25041">issues.joomla.org/tracker/joomla-cms/25041</a>.</sub>

**Comment by roland-d** [comment]
Thank you.

**Pull request merged**